### ISSUE TITLE
Coroot UI fails to see one of postgres instances.

### ISSUE THREAD

**Issue by Compunctus** [coroot/coroot#15]
Hi. Coroot doesn't show one of the replicas in my 3-server postgres setup as a postgres node. Also it fails to see replication roles, and doesn't recognize other deployments (gitlab/nextcloud) as postgres.

![image](https://user-images.githubusercontent.com/5058853/211399685-4f78e11d-d91e-4478-8ff1-9464130a044d.png)

The relevant metrics are present in prometheus:
![image](https://user-images.githubusercontent.com/5058853/211399912-c671dcd0-6223-4d03-b101-0cfcd3b33666.png)

![image](https://user-images.githubusercontent.com/5058853/211399951-4cd40d30-a6f3-4d39-a0d0-a4f3e7814623.png)

![image](https://user-images.githubusercontent.com/5058853/211400053-65bb6a0b-530f-4456-ba51-1734d0ed9901.png)

Could you please advise on how to handle this? I get the feeling that coroot doesn't like low-query situations or my labels.

**Comment by def** [comment]
Is `psqlha-postgresql-ha-postgresql-2` displayed on the `Instances` tab?
Can you also check the Coroot's log for errors?

Replication roles are currently discovered only for clusters managed by the k8s operators for Postgres (Zalando, CrunchyData, Percona).

**Comment by Compunctus** [comment]
Yep, it's displayed on instances tab. All three are green on the instances tab, now psqlha-postgresql-ha-postgresql-1 is not recognised as a postgres service.

I do see new errors in logs, namely:
```
couldn't find actual instance for "postgres", initial instance is "10.1.18.172:80" (map[])
```

and that ip (18.172) does belong to pod psqlha-postgresql-ha-postgresql-1

**Comment by Compunctus** [comment]
Both metrics were empty. I also noticed that they were simply wrong for most IP addresses - showing ports that could never exist in a pod. Restarted daemonset, now those metrics seem valid. Waiting for 10 minutes to let coroot settle.

No noticeable errors were found in node-agent logs prior to restarting the daemonset.
